mlflow/pytorch/torchserve/deploy/serve.py

```diff
@@ -0,0 +1,388 @@
+            raise Exception(
+                "Unable to infer the results for the name %s. Server returned status code %s and response: %s"
+                % (deployment_name, resp.status_code, resp.content)
+            )
+
+        return resp.text
+
+    def __generate_mar_file(
+        self, model_name, version, model_file, handler_file, extra_files, model_uri
+    ):
+
+        """
+        Generates mar file using the torch archiver in the specified model store path
+        """
+        valid_file_suffixes = [".pt", ".pth"]
+        requirements_file = "requirements.txt"
+        req_file_path = None
+
+        if not os.path.isfile(model_uri):
+            path = Path(_download_artifact_from_uri(model_uri))
+            model_config = path / "MLmodel"
+            if not model_config.exists():
+                raise Exception(
+                    "Failed to find MLmodel configuration within the specified model's root directory."
+                )
+            else:
+                model_path = None
+                if path.suffix in valid_file_suffixes:
+                    model_uri = path
+                else:
+                    for file in path.iterdir():
+                        if file.is_dir():
+                            for sub_files in file.iterdir():
+                                if sub_files.suffix in valid_file_suffixes:
+                                    model_path = sub_files
+                                if PurePath(sub_files).name == requirements_file:
+                                    req_file_path = sub_files
+                        else:
+                            if file.suffix in valid_file_suffixes:
+                                model_path = file
+                            if PurePath(file).name == requirements_file:
+                                req_file_path = file
+                    if model_path is None:
+                        raise RuntimeError(
+                            "Model file does not have a valid suffix. Expected to be one of "
+                            + ", ".join(valid_file_suffixes)
+                        )
+                    model_uri = model_path
+
+        export_path = self.server_config["export_path"]
+        if export_path:
+            model_store = export_path
+        else:
+            model_store = "model_store"
+            if not os.path.isdir(model_store):
+                os.makedirs(model_store)
+
+        cmd = (
+            "torch-model-archiver --force --model-name {} "
+            "--version {} --model-file {} --serialized-file {} "
+            "--handler {} --export-path {}".format(
+                model_name, version, model_file, model_uri, handler_file, model_store
+            )
+        )
+        if extra_files:
+            cmd = "{} --extra-files {}".format(cmd, extra_files)
+        if req_file_path:
+            cmd = "{} -r {}".format(cmd, req_file_path)
+
+        return_code = os.system(cmd)
```

Comment:
Can we use `subprocess.run` to run this command instead? Usage of the `subprocess` APIs is encouraged over the older `os.system` API: https://docs.python.org/3/library/subprocess.html#subprocess.run